crates/services/txpool/src/error.rs

```diff
@@ -0,0 +1,166 @@
+use fuel_core_types::{
+    fuel_tx::{
+        Address,
+        BlobId,
+        ContractId,
+        TxId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+    fuel_vm::checked_transaction::CheckError,
+};
+
+use crate::ports::WasmValidityError;
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+#[allow(missing_docs)]
+#[derive(Debug, Clone, derive_more::Display)]
+#[non_exhaustive]
+pub enum Error {
+    #[display(fmt = "Gas price not found for block height {_0}")]
+    GasPriceNotFound(String),
+    #[display(fmt = "Failed to calculate max block bytes: {_0}")]
+    MaxBlockBytes(String),
+    #[display(fmt = "TxPool required that transaction contains metadata")]
+    NoMetadata,
+    #[display(fmt = "TxPool doesn't support this type of transaction.")]
+    NotSupportedTransactionType,
+    #[display(fmt = "Transaction is not inserted. Hash is already known")]
+    NotInsertedTxKnown,
+    #[display(
+        fmt = "Transaction is not inserted. Pool limit is hit, try to increase gas_price"
+    )]
+    NotInsertedLimitHit,
+    #[display(
+        fmt = "Transaction is not inserted. Attempting to upgrade WASM bytecode when WASM support is not enabled."
+    )]
+    NotInsertedWasmNotEnabled,
+    #[display(
+        fmt = "Transaction is not inserted. WASM bytecode matching the given root was not found."
+    )]
+    NotInsertedWasmNotFound,
+    #[display(
+        fmt = "Transaction is not inserted. WASM bytecode contents are not valid."
+    )]
+    NotInsertedInvalidWasm,
+    #[display(fmt = "Transaction is not inserted. The gas price is too low.")]
+    NotInsertedGasPriceTooLow,
+    #[display(
+        fmt = "Transaction is not inserted. More priced tx {_0:#x} already spend this UTXO output: {_1:#x}"
+    )]
+    NotInsertedCollision(TxId, UtxoId),
+    #[display(
+        fmt = "Transaction is not inserted. More priced tx has created contract with ContractId {_0:#x}"
+    )]
+    NotInsertedCollisionContractId(ContractId),
+    #[display(
+        fmt = "Transaction is not inserted. More priced tx has uploaded the blob with BlobId {_0:#x}"
+    )]
+    NotInsertedCollisionBlobId(BlobId),
+    #[display(
+        fmt = "Transaction is not inserted. A higher priced tx {_0:#x} is already spending this message: {_1:#x}"
+    )]
+    NotInsertedCollisionMessageId(TxId, Nonce),
+    #[display(fmt = "Transaction is not inserted. UTXO input does not exist: {_0:#x}")]
+    NotInsertedOutputDoesNotExist(UtxoId),
```

Comment:
This error does not match the display string ~

```suggestion
    #[display(fmt = "Transaction is not inserted. UTXO input does not exist: {_0:#x}")]
    NotInsertedInputDoesNotExist(UtxoId),
```

Reply:
seems to be the same as `NotInsertedInputDoesNotExist(UtxoId)` then, which is defined below?

Reply:
This also seems unused so I think I can just remove the variant. Thank you for spotting! 🙏

Reply:
Done! ✅